src/symbol/collision_feature.js

```diff
@@ -72,42 +72,60 @@ class CollisionFeature {
     _addLineCollisionBoxes(collisionBoxArray, line, anchor, segment, labelLength, boxSize, featureIndex, sourceLayerIndex, bucketIndex) {
         const step = boxSize / 2;
         const nBoxes = Math.floor(labelLength / step);
+        // We calculate line collision boxes out to 150% of what would normally be our
+        // max size, to allow collision detection to work on labels that expand as
+        // they move into the distance
+        const nPitchPaddingBoxes = Math.floor(nBoxes / 4);
```

Comment:
Yeah, this one adds collision detection cost even in the un-pitched state because these extra boxes get returned in the `CollisionTile` grid query.

The other place where collision detection gets more expensive is in `CollisionTile::placeCollisionFeature`, where we expand the size of our grid query based on the "perspective ratio". That change, however, only really affects tiles that are pitched/distant.